sgx/epid-types/src/quote_nonce.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+//! Quote nonce wrapper
+
+use mc_sgx_core_types::impl_ffi_wrapper;
+use mc_sgx_epid_types_sys::sgx_quote_nonce_t;
+use mc_util_from_random::FromRandom;
+use rand_core::{CryptoRng, RngCore};
+
+/// The size of a [QuoteNonce] structure's x64 representation, in bytes.
+pub const QUOTE_NONCE_SIZE: usize = 16;
+
+/// A structure wrapping a nonce to be used in an SGX quote
+///
+/// # Example
+///
+/// ```
+/// use mc_sgx_epid_types::QuoteNonce;
+/// use mc_util_from_random::FromRandom;
+/// use rand_core::SeedableRng;
+/// use rand_hc::Hc128Rng;
+///
+/// // chosen by fair dice roll, or: use a real rng in real code, folks.
+/// let mut csprng = Hc128Rng::seed_from_u64(0);
+/// let nonce = QuoteNonce::from_random(&mut csprng);
+/// let nonce_contents: &[u8] = nonce.as_ref();
+/// let expected = [226u8, 30, 184, 201, 207, 62, 43, 114, 89, 4, 220, 27, 84, 79, 238, 234];
+/// assert_eq!(nonce_contents, &expected[..]);
+/// ```
+#[derive(Default)]
+#[repr(transparent)]
+pub struct QuoteNonce(sgx_quote_nonce_t);
+
+impl_ffi_wrapper! {
+    QuoteNonce, sgx_quote_nonce_t, QUOTE_NONCE_SIZE, rand;
+}
+
+impl FromRandom for QuoteNonce {
```

Comment:
Where do we currently verify the QuoteNonce? I recall discussions around verifying the quote nonce several months ago, so I apologize if I'm forgetting.

Reply:
https://github.com/mobilecoinofficial/mobilecoin/blob/633a33b2c4ec2709c345482dfd33ced1fe1e183a/crypto/ake/enclave/src/lib.rs#L349

Reply:
🙏